core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -58,7 +81,13 @@ public TableScan newScan() {
   @Override
   public Schema schema() {
     if (table().spec().fields().size() < 1) {
-      return schema.select("record_count", "file_count");
```

Comment:
I think there's actually a bug here, and it only checks latest spec for Unpartitioned.  if we have partition fields before but removed them, we will not show them.  See other metadata tables like BaseFilesTable.schema.  

Anyway its unrelated, but made https://github.com/apache/iceberg/issues/7533 to track it.

Reply:
Good catch. I will explore this in the follow-up.